### DISCUSSION TITLE
3 environment when attempting to compile?

### DISCUSSION THREAD

**Discussion by wwian** [mriscoc/Ender3V2S1#308] [Q&A] [answered]
Today I opened Ender3V2S1-Released in Visual Studio and updated the config to Ender 3 S1. Using 'Auto Build Marlin', I attempted to compile and was presented with 3 environments to choose from. I have searched online for info on jlink & stlink. However, I can't find any references. Can someone share a pearl of wisdom as to which environment is the correct one? Thanks!

![image](https://user-images.githubusercontent.com/25714748/186290122-d033f74f-02fa-4161-b6c0-2e244d8da9d8.png)

**Comment by mriscoc** [comment, marked as answer]
Use the first one: `STM32F401RC_creality`

**Reply by wwian** [reply]
Thank you! Professional Firmware is so awesome!
